### PR TITLE
[feat] engine: implementation of wallhaven

### PR DESCRIPTION
**Pull request by Bnyro**
## What does this PR do?
* adds a new engine for images: wallhaven
* it's enabled by default because the results are really good and the response times short

## Why is this change important?
* because the engine is very useful if searching for anything nature-related

## How to test this PR locally?
* !wallhaven sunset

### CODE REVIEW

File: searx/engines/wallhaven.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    'require_api_key': False,
+    'results': 'JSON',
+}
+categories = ['images']
+paging = True
+
+base_url = "https://wallhaven.cc"
+
+# using an api key is optional and will increase the rate limit
+api_key = ''
```
Comment (return42) [inline searx/engines/wallhaven.py]: @Bnyro do you know where the rate limits are?

Comment (Bnyro) [reply searx/engines/wallhaven.py]: 45 requests per minute, see https://wallhaven.cc/help/api#limits

Comment (Bnyro) [reply searx/engines/wallhaven.py]: It's possible that the API key doesn't actually impact the rate limit, not sure about that. In their docs, it only says that the api key would be used for users to access "some of their settings" ...

**Review by return42 — APPROVED** [review]
> it's enabled by default because the results are really good and the response times short

Yes, engine is very fast .. good default addition to the image category.

FYI: I squashed some doc into your commit [diff](https://github.com/searxng/searxng/compare/384b2d002d5016f6200e8ae87301ee8f1df8e607..7b9aefe0063e196c34d8c8b62a9b48ffb7ff3c46)

**Comment by Bnyro** [comment]
Thanks for the addition!

Indeed, the description of sfw, sketchy & nsfw is worth having its place in the docs :joy:

**Pull request merged**